### DISCUSSION TITLE
Best practice for routing and initializing client in serve

### DISCUSSION THREAD

**Discussion by Alex-Wenner-FHR** [h2oai/wave#848] [Q&A]
Currently how I am doing this. I know this is probably wrong, I get errors when I try to navigate to 'le_view' and I assume it stems from this. Any recommendations on how to best do this would be helpful!
![image](https://user-images.githubusercontent.com/78372056/119890395-670f6e80-befd-11eb-9bf3-d20d66f5d729.png)

**Comment by lo5** [comment]
It's missing a `q.client.initialized = True` inside the `if not`? Also, the page will be saved twice if it hits any of the inner conditionals.

**Reply by Alex-Wenner-FHR** [reply]
I have a q.client.initalized = True inside my init_client function, would that not suffice setting that to True?

**Reply by lo5** [reply]
What is not working, exactly? What are the errors you are running into?

**Reply by Alex-Wenner-FHR** [reply]
Now I am not getting any errors, my client nor app seems to stay initialized. It will render to the 'home' page after client and app have been initialized. Then when I attempt to re route, it will re initialize both and re render the 'home' page.

**Reply by Alex-Wenner-FHR** [reply]
Here are my client and app initializations by the way 
![image](https://user-images.githubusercontent.com/78372056/119922679-eb7de380-bf35-11eb-9c4d-79ac4790f8c3.png)
